src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/blobs/blob.rs

```diff
@@ -1,15 +1,39 @@
 use crate::prelude::*;
 
 /// Blob is a wrapper a bag of bytes
-#[derive(Clone, PartialEq, Eq, Debug, derive_more::Display, uniffi::Record)]
+#[derive(
+    Clone,
+    PartialEq,
+    Eq,
+    DeserializeFromStr,
+    SerializeDisplay,
+    uniffi::Record,
+    derive_more::Display,
+    derive_more::Debug,
+)]
+#[display("{}", self.to_hex())]
+#[debug("{}", self.to_hex())]
 pub struct Blob {
     pub(crate) secret_magic: BagOfBytes,
 }
 
+impl FromStr for Blob {
```

Comment:
I think you can remove "manual" impl of `FromStr` and use `derive_more::FromStr`